lib/travis/cli/repo_command.rb

```diff
@@ -83,6 +82,14 @@ def detect_slug
           end
         end
 
+        def parse_remote(url)
+          URI.parse(url).path
+        rescue URI::InvalidURIError
```

Comment:
I am against using exceptions as a part of the flow control. If `git@…` is an expected "URI", we should handle it as such.